### ISSUE TITLE
How to not use the VPN default gateway for internet?

### ISSUE THREAD

**Issue by uniquegodwin** [pivpn/pivpn#294]
Hello,

How do I stop internet access through the VPN and use VPN only to access remote network resources?

Thanks.

**Comment by cfcolaco** [comment]
hmm, I have never done such thing, guess we must do some tests and try some options, and we both going to learn from this! :P 

first, try commenting 

`push "redirect-gateway def1"` 

which is responsible for redirecting all traffic trough VPN

**Comment by harmanbirdi** [comment]
I tried commenting out the <code>push redirect-gateway def1</code> line but then
the VPN part stopped working for me. I could not connect to any of my clients. Am I missing something here.

I would like only the VPN traffic to be sent through the VPN tunnel. Any pointers to
resolve this would be greatly appreciated.

Here is what I see - first with the def1 line

```
$ $ ifconfig | grep 10.8
	inet 10.8.0.2 --> 10.8.0.2 netmask 0xffffff00
$ traceroute www.yahoo.com
traceroute: Warning: www.yahoo.com has multiple addresses; using 206.190.39.43
traceroute to atsv2-fp.wg1.b.yahoo.com (206.190.39.43), 64 hops max, 52 byte packets
 1  10.8.0.1 (10.8.0.1)  246.142 ms  246.679 ms  243.987 ms
 2  * * * *^C
$ ping www.yahoo.com
PING atsv2-fp.wg1.b.yahoo.com (106.10.178.36): 56 data bytes
64 bytes from 106.10.178.36: icmp_seq=0 ttl=41 time=628.971 ms
^C
--- atsv2-fp.wg1.b.yahoo.com ping statistics ---
2 packets transmitted, 1 packets received, 50.0% packet loss
round-trip min/avg/max/stddev = 628.971/628.971/628.971/0.000 ms
$ ping 10.0.102.4
PING 10.0.102.4 (10.0.102.4): 56 data bytes
64 bytes from 10.0.102.4: icmp_seq=0 ttl=63 time=454.736 ms
64 bytes from 10.0.102.4: icmp_seq=1 ttl=63 time=258.390 ms
^C
--- 10.0.102.4 ping statistics ---
2 packets transmitted, 2 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 258.390/356.563/454.736/98.173 ms
$ 
```

This one is without def1 line.

```
$ ifconfig | grep 10.8
	inet 10.8.0.2 --> 10.8.0.2 netmask 0xffffff00
$ $ traceroute www.yahoo.com
traceroute: Warning: www.yahoo.com has multiple addresses; using 206.190.39.42
traceroute to atsv2-fp.wg1.b.yahoo.com (206.190.39.42), 64 hops max, 52 byte packets
 1  192.168.123.1 (192.168.123.1)  2.959 ms  2.498 ms  2.555 ms
 2  * * * *
^C
$ ping www.yahoo.com
PING atsv2-fp.wg1.b.yahoo.com (206.190.39.43): 56 data bytes
64 bytes from 206.190.39.43: icmp_seq=0 ttl=50 time=50.183 ms
^C
--- atsv2-fp.wg1.b.yahoo.com ping statistics ---
1 packets transmitted, 1 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 50.183/50.183/50.183/0.000 ms
$ ping 10.0.102.4
PING 10.0.102.4 (10.0.102.4): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
^C
--- 10.0.102.4 ping statistics ---
3 packets transmitted, 0 packets received, 100.0% packet loss
$ 
```

**Comment by cfcolaco** [comment]
@harmanbirdi yes ... well .. the good part is ... no traffic to the internet!! ... Now I ask for some patience so I can better look at it when I have some time ... which right now I don't have time to dedicate to all the issues ant things around here! so it can take a bit more time! if you do any findings please keep this updated please.

**Comment by harmanbirdi** [comment]
@cfcolaco - I did manage to resolve this. After taking a detailed look at the routing tables, before  and after VPN connection - I realized that the route I was pushing from the server was incorrect. My local subnet route being pushed should have been (notice the subnet mask).

```
push "route 10.0.0.0 255.255.0.0"
```

instead of

```
push "route 10.0.0.0 255.255.255.0"
```

Once I made that change and commented out the <code>push "redirect-gateway def1"</code>,
things started working fine. Now, my VPN traffic flows through the VPN gateway, and rest flows
through my normal internet gateway.

**Comment by cfcolaco** [comment]
Awesome!! so I was correct after all!! =) Good to know .. marking this as information .. and closing the issue! if you need help with something else just ask .. I will help as soon as I can! 

Thank you for your feedback!!

**Comment by narbik** [comment]
I know this was closed but i need Exactly what @harmanbirdi did, done. I do not understand where the push "route 10.0.0.0 255.255.255.0" goes. please help. thanks.

**Comment by harmanbirdi** [comment]
Hello @narbik - This information would go into your /etc/openvpn/server.conf file.  Hope that helps.

**Comment by narbik** [comment]
thank you @harmanbirdi !
When I add 
push "route 192.168.0.0 255.255.255.0" 
and comment out 
push "redirect-gateway def1"
no clients will connect to the server.  i receive the following error
Wed Aug 22 18:49:05 2018 read UDP: Unknown error (code=10054)

**Comment by harmanbirdi** [comment]
@narbik - Its not that simple. I would need to know how your network is setup. Is your router
allowing UDP traffic in for the port that is being used to connect. If not, do update your firewall
to allow that traffic through. What is the internal network and its subnet that is being used?  
The route that you are pushing to should be the one that is in your router - likely serving as your DHCP.

Hope you have solved it by now.

**Comment by narbik** [comment]
@harmanbirdi I am still working on this issue as i have put out 3 different PiVPN Solutions now. they all work well except they are routing the clients internet which i do not want. if the local subnet is 192.168.0.0/24 i would assume i am adding 
`push "route 192.168.0.0 255.255.255.0"`
and commenting out 
`push "redirect-gateway def1"`

but when i do that no clients can connect. I still receive:
` read UDP: Unknown error (code=10054)`

when i comment out the push route and remove the comment on redirect it works fine...
I would love to just see an option during the install that asks weather you want internet routing or not.
